### PR TITLE
gitserver: Remove repos-stats.json file

### PR DESCRIPTION
**Pull request by eseliger**
The content of this file is the size of all repos combined per shard. We update it in the janitor after all repos are processed, but at the same time we collect the same sizes per repo and write them to gitserver_repos.repo_size_bytes. This didn't exist back when the file was introduced, but nowadays, they are pretty much equal. An additional benefit is that other processes that update the repo size in the DB will also reflect faster in the value read here. This removes a bunch of complexity around managing that file, maintaining endpoints to retrieve that data, and around talking to every gitserver to get the cumulated value.

Proof: On dotcom, where 1 gitserver holds roughly 15.5TB of repo data, the calculated value in the file and the value from the DB only differ by 175MB (0.001076729067%).

**Please let me know if I am missing some nuance here and those numbers are not actually the same!**

With quarterly releases, we can only remove some of the remaining endpoints code in 5.3, to keep compatibility.

There are exactly two things that use this endpoint:
  - A GraphQL resolver that returns the total git size
    - If the frontend comes up first, it just uses the DB, which is already populated correctly long ago
    - If the new gitserver comes up first, it will still have the endpoints but read from the DB, so no problem here
  - The usagestats code, that collects pings data regularly
    - If the frontend comes up first, this is not a problem as the DB is already populated correctly and it's read from there
    - If the gitserver comes up first, it will still have the endpoints but read from the DB, so no problem here

## Test plan

Adjusted existing test, and collected proof on Sourcegraph.com that the numbers are virtually the same. Also ran new gitservers locally against a main frontend, to see that things don't break.

### CODE REVIEW

File: cmd/gitserver/server/cleanup_test.go [hunk 1]
```diff
@@ -47,6 +44,7 @@ func newMockedGitserverDB() database.DB {
 	return db
 }
 
+// TODO: Only test the repo size part of the cleanup routine, not all of it.
```
Comment (mrnugget) [inline cmd/gitserver/server/cleanup_test.go:47]: Is this for later or for this PR?

File: cmd/gitserver/server/repo_info.go [hunk 2]
```diff
@@ -3,42 +3,44 @@ package server
...
-		// yet. Clients are expected to handle this case by noticing UpdatedAt
-		// is not set.
-		b = []byte("{}")
-	} else if err != nil {
-		return nil, err
+	resp := protocol.ReposStats{
+		UpdatedAt: time.Now(), // Unused value, to keep the API pretend the data is fresh.
+		// Divide the size by shard count so that the cumulative number on the client
+		// side is correct again.
+		GitDirBytes: size / int64(shardCount),
```
Comment (mrnugget) [inline cmd/gitserver/server/repo_info.go:34]: Why not give `GetGitserverGitDirSize` an optional `shardID` parameter that can be passed in if the size for a given shard is required?

**Review by mrnugget — APPROVED** [review]
Makes sense, IMHO

File: cmd/gitserver/server/cleanup_test.go (hunk 1 above)
Comment (eseliger) [reply cmd/gitserver/server/cleanup_test.go:47]: Was thinking for later

File: cmd/gitserver/server/repo_info.go (hunk 2 above)
Comment (mrnugget) [reply cmd/gitserver/server/repo_info.go:34]: Okay, if you think it's a non-issue, then that's fine too.

**Pull request merged**